pandas/core/groupby/generic.py

```diff
@@ -834,9 +833,10 @@ def _wrap_output(self, output, index, names=None):
             return Series(output, index=index, name=name)
 
     def _wrap_aggregated_output(self, output, names=None):
-        return self._wrap_output(output=output,
-                                 index=self.grouper.result_index,
-                                 names=names)
+        result = self._wrap_output(output=output,
+                                   index=self.grouper.result_index,
+                                   names=names)
+        return self._reindex_output(result)._convert(datetime=True)
```

Comment:
if you want to try and clean this up would be great (just see how much closer we can get to a common _wrap_*) that is more common for DataFrame/Series

e.g. first thing I would do is move _reindex_output(...)._convert(...) *into* _wrap_output

but see if you majorly break things, there are some special case paths which are not very friendly.

*could* do this as a followup as well.

Reply:
That was my first instinct as well, however:

1. Isn't `transform` supposed to keep the shape intact? This might mean that `observed` kwarg (whose functionality is, depending on the passed value, supposed to change the shape of the data) and `transform` are incompatible in general.
2. I was drawing inspiration from the existing implementation in `DataFrameGroupBy` which uses `_reindex_output` in `wrap_applied_output` and `wrap_aggregated_output`, but not `wrap_transformed_output`. `DataFrameGroupBy` also has disjoint implementations for `wrap_aggregated_output`, `wrap_applied_output` and `_wrap_transformed_output`, so it feels like `SeriesGroupBy`'s common wrap usage for `wrap_aggregated_output` and `_wrap_transformed_output` up until now has been just a nice coincidence.
3. I tried this change right now and it does make some `transform` related tests fail. It doesn't look major, but I am still not sure we should be mixing `transform` and reindexing (or adding `nan` rows for unobserved groups).